From d7bac55b121ae8913d6826d90b305a317394f526 Mon Sep 17 00:00:00 2001
From: Elvis Pranskevichus <elvis@magic.io>
Date: Tue, 17 Jul 2018 17:05:01 -0400
Subject: [PATCH] Use environment marker for typing dependency instead of
 runtime

Python version constraints *must* be specified as environment markers.
Runtime setup.py checks will not be reflected in the metadata, resulting
in incorrect hard dependency on typing in dist-info and in wheels.
---
 setup.py | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/setup.py b/setup.py
index 4232cf4..7feff23 100644
--- a/setup.py
+++ b/setup.py
@@ -31,9 +31,10 @@ classifiers = [
     'Topic :: Software Development',
 ]
 
-install_requires = ['mypy_extensions >= 0.3.0']
-if sys.version_info < (3, 5):
-    install_requires.append('typing >= 3.6.1')
+install_requires = [
+    'mypy_extensions >= 0.3.0',
+    'typing >= 3.6.1;python_version<"3.5"'
+]
 
 setup(
     name='typing_inspect',
-- 
2.17.1

